### PR TITLE
Add JSDoc to deno_typescript

### PR DESCRIPTION
**Pull request by kitsonk**
This PR adds JSDoc to the JavaScript in `deno_typescript`, which can be used with VSCode to help ensure the type safety of the JavaScript code without it being written in TypeScript.  This is a stepping stone to potentially shared parts between `deno_typescript` and the Deno runtime compiler.

### CODE REVIEW

File: deno_typescript/compiler_main.js [hunk 1]
```diff
@@ -216,37 +246,18 @@ class Host {
...
+ * @param {string} configurationText
+ */
 function configure(configurationText) {
   const { config, error } = ts.parseConfigFileTextToJson(
     "tsconfig.json",
     configurationText
   );
   if (error) {
-    return { diagnostics: [error] };
+    return { options: {}, diagnostics: [error] };
```
Comment (kitsonk) [inline deno_typescript/compiler_main.js]: deals with `options` possibly being undefined in the return

File: deno_typescript/compiler_main.js [hunk 2]
```diff
@@ -270,14 +285,21 @@ function dispatch(opName, obj) {
   return res["ok"];
 }
 
+/**
+ * @param {number} code
+ */
 function exit(code) {
   dispatch("exit", { code });
-  unreachable();
+  return unreachable();
```
Comment (kitsonk) [inline deno_typescript/compiler_main.js]: Done so the return type is `never`.

File: deno_typescript/globals.d.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+// This scopes the `ts` namespace globally, which is where it exists at runtime
+// when building Deno, but the `typescript/lib/typescript.d.ts` is defined as a
+// module
+
+import * as _ts from "typescript";
+
+declare global {
+  namespace ts {
+    export = _ts;
```
Comment (ry) [inline deno_typescript/globals.d.ts]: Where is this file used? 

I wonder if you've seen this hack I have in `deno_typescript/compiler_main.js`
https://github.com/denoland/deno/blob/718c8c4360b5c42d1c0bcd97d7348908d915212e/deno_typescript/compiler_main.js#L141-L144
basically the `typescript.d.ts` which ships with typescript ends in `export = ts;` and for whatever reason `getSourceFile` doesn't like that...

Comment (kitsonk) [reply deno_typescript/globals.d.ts]: I saw it, but didn't touch it.  I suspect it is doing the same thing I am doing.  What the actual file does, is if it isn't loaded as a CommonJS module, it hoists itself into the global as `ts`, the problem is the `.d.ts` file only expresses it as module.  So to remove the hack we need to essentially do this for the runtime TypeScript/JavaScript.  Let me try to fix that all together then.

**Pull request merged**